tensorflow_lite_support/c/test/task/vision/image_classifier_test.cc

```diff
@@ -0,0 +1,302 @@
+ protected:
+  void SetUp() override {
+    TfLiteImageClassifierOptions options = {0};
+    options.base_options.model_file.file_path =
+        JoinPath("./" /*test src dir*/, kTestDataDirectory,
+                 kMobileNetQuantizedWithMetadata)
+            .data();
+    image_classifier = TfLiteImageClassifierFromOptions(&options);
+
+    ASSERT_NE(image_classifier, nullptr);
+  }
+
+  void TearDown() override { TfLiteImageClassifierDelete(image_classifier); }
+  TfLiteImageClassifier* image_classifier;
+};
+
+TEST_F(CImageClassifierClassifyTest, SucceedsWithImageData) {
+  CImageData image_data = LoadImage("burger-224.png");
+
+  TfLiteFrameBuffer frame_buffer = {.dimension.width = image_data.width,
+                                    .dimension.height = image_data.height,
+                                    .buffer = image_data.pixel_data,
+                                    .format = kRGB};
+
+  TfLiteClassificationResult* classification_result =
+      TfLiteImageClassifierClassify(image_classifier, &frame_buffer);
+
+  CImageDataFree(&image_data);
+
+  ASSERT_NE(classification_result, nullptr) << "Classification Result is NULL";
+  EXPECT_TRUE(classification_result->size >= 1)
+      << "Classification Result size is 0";
+  EXPECT_NE(classification_result->classifications, nullptr)
+      << "Classification Result Classifications is NULL";
+  EXPECT_TRUE(classification_result->classifications->size >= 1)
+      << "Classification Result Classifications Size is 0";
+  EXPECT_NE(classification_result->classifications->categories, nullptr)
+      << "Classification Result Classifications Classes is NULL";
+
+  TfLiteClassificationResultDelete(classification_result);
+}
+
+TEST_F(CImageClassifierClassifyTest, SucceedsWithRoiWithinImageBounds) {
+  CImageData image_data = LoadImage("burger-224.png");
+
+  TfLiteFrameBuffer frame_buffer = {.dimension.width = image_data.width,
+                                    .dimension.height = image_data.height,
+                                    .buffer = image_data.pixel_data,
+                                    .format = kRGB};
+
+  TfLiteBoundingBox bounding_box = {
+      .origin_x = 0, .origin_y = 0, .width = 100, .height = 100};
+  TfLiteClassificationResult* classification_result =
+      TfLiteImageClassifierClassifyWithRoi(image_classifier, &frame_buffer,
+                                           &bounding_box);
+
+  CImageDataFree(&image_data);
+
+  ASSERT_NE(classification_result, nullptr) << "Classification Result is NULL";
+  EXPECT_TRUE(classification_result->size >= 1)
+      << "Classification Result size is 0";
+  EXPECT_NE(classification_result->classifications, nullptr)
+      << "Classification Result Classifications is NULL";
+  EXPECT_TRUE(classification_result->classifications->size >= 1)
+      << "Classification Result Classifications Size is 0";
+  EXPECT_NE(classification_result->classifications->categories, nullptr)
```

Comment:
Need to check the score and labels. Same for the other "Succeeds" tests. Again, OK to add the tests in another PR. Please add a TODO here.

Reply:
Added TODOs